flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkResource.java

```diff
@@ -38,6 +39,8 @@
 	 */
 	void addConfiguration(Configuration config) throws IOException;
 
+	FlinkDistribution getDistribution();
```

Comment:
This will relatively soon not make sense, because in distributed settings there could be multiple distributions that need to be updated.

Stay tuned, I'm still figuring things out on this front.